buildSrc/src/main/kotlin/project/java.gradle.kts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package project
+
+plugins {
+    id("java-library")
+    id("checkstyle")
+    id("project.base")
+    id("conventions.bnd")
+    id("conventions.javadoc")
+    id("conventions.optional")
+    id("conventions.publishing")
+    id("conventions.spotbugs")
+    id("conventions.spotless")
+    id("conventions.testing-junit")
+}
+
+dependencies { "optionalApi"(libs.slf4j) }
```

Comment:
Remind me what "optionalApi" means?  I would think this is implementation, not api.

Reply:
I had it as "optionalImplementation" but then it changes the scope to runtime in the pom.xml files and prior to this change it was compile. 

This keeps it the same as it has been even though I don't believe we surface and SLF4J API to userland code.